workers_domain.go

```diff
@@ -0,0 +1,126 @@
+package cloudflare
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+)
+
+var (
+	ErrMissingHostname    = errors.New("required hostname missing")
+	ErrMissingService     = errors.New("required service missing")
+	ErrMissingEnvironment = errors.New("required environment missing")
+)
+
+type WorkersDomain struct {
+	ID          string `json:"id,omitempty"`
+	ZoneID      string `json:"zone_id,omitempty"`
+	ZoneName    string `json:"zone_name,omitempty"`
+	Hostname    string `json:"hostname,omitempty"`
+	Service     string `json:"service,omitempty"`
+	Environment string `json:"environment,omitempty"`
+}
+
+type WorkersDomainResponse struct {
+	Response
+	Result WorkersDomain `json:"result"`
+}
+
+type WorkersDomainListParams struct {
+	ZoneID      string `url:"zone_id,omitempty"`
+	ZoneName    string `url:"zone_name,omitempty"`
+	Hostname    string `url:"hostname,omitempty"`
+	Service     string `url:"service,omitempty"`
+	Environment string `url:"environment,omitempty"`
+}
+
+type WorkersDomainListResponse struct {
+	Response
+	Result []WorkersDomain `json:"result"`
+}
+
+// WorkersListDomains lists all Worker Domains.
+//
+// API reference: https://api.cloudflare.com/#worker-domain-list-domains
+func (api *API) WorkersListDomains(ctx context.Context, rc *ResourceContainer, params WorkersDomainListParams) ([]WorkersDomain, error) {
```

Comment:
we should have the verb followed by the noun so here, `List` (verb) `WorkersDomains` (noun)
```suggestion
func (api *API) ListWorkersDomains(ctx context.Context, rc *ResourceContainer, params WorkersDomainListParams) ([]WorkersDomain, error) {
```